BUG: GPUTreeExplainer tests failing with Broadcasting error

### Issue Description

Running `python -m pytest tests/explainers/test_gpu_tree.py` on my old RTX 2080Ti results in multiple failures.

### Minimal Reproducible Example

```python
# run on CUDA capable hardware
python setup.py build_ext --inplace  # build CUDA extensions
python -m pytest tests/explainers/test_gpu_tree.py
```

### Traceback

```shell
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap[interventional-xgboost.sklearn.XGBClassifier1] - ValueError: operands could not be broadcast together with shapes (100,4,5) (5,100,4)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap[interventional-lightgbm.sklearn.LGBMClassifier1] - ValueError: operands could not be broadcast together with shapes (100,4,5) (5,100,4)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap[interventional-sklearn.ensemble._forest.RandomForestClassifier0] - ValueError: operands could not be broadcast together with shapes (100,4,2) (2,100,4)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap[interventional-sklearn.ensemble._forest.RandomForestClassifier1] - ValueError: operands could not be broadcast together with shapes (100,4,5) (5,100,4)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap[tree_path_dependent-xgboost.sklearn.XGBClassifier1] - ValueError: operands could not be broadcast together with shapes (100,4,5) (5,100,4)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap[tree_path_dependent-lightgbm.sklearn.LGBMClassifier1] - ValueError: operands could not be broadcast together with shapes (100,4,5) (5,100,4)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap[tree_path_dependent-sklearn.ensemble._forest.RandomForestClassifier0] - ValueError: operands could not be broadcast together with shapes (100,4,2) (2,100,4)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap[tree_path_dependent-sklearn.ensemble._forest.RandomForestClassifier1] - ValueError: operands could not be broadcast together with shapes (100,4,5) (5,100,4)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap_interactions[tree_path_dependent-xgboost.sklearn.XGBClassifier1] - ValueError: operands could not be broadcast together with shapes (4,100) (5,)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap_interactions[tree_path_dependent-lightgbm.sklearn.LGBMClassifier1] - ValueError: operands could not be broadcast together with shapes (4,100) (5,)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap_interactions[tree_path_dependent-sklearn.ensemble._forest.RandomForestClassifier0] - ValueError: operands could not be broadcast together with shapes (4,100) (2,)
FAILED tests/explainers/test_gpu_tree.py::test_gpu_tree_explainer_shap_interactions[tree_path_dependent-sklearn.ensemble._forest.RandomForestClassifier1] - ValueError: operands could not be broadcast together with shapes (4,100) (5,)
================================================================ 12 failed, 23 passed, 2 xfailed, 5 warnings in 14.00s =================================================================
```

### Expected Behavior

_No response_

### Bug report checklist

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [x] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

master

--- a/tests/explainers/test_gpu_tree.py
+++ b/tests/explainers/test_gpu_tree.py
@@ -231,6 +231,11 @@ def test_gpu_tree_explainer_shap(task, feature_perturbation):
     host_shap = ex.shap_values(X, check_additivity=True)
     gpu_shap = gpu_ex.shap_values(X, check_additivity=True)
 
+    # todo: this should actually happen in the GPUTreeExplainer
+    if np.array(gpu_shap).ndim == 3:
+        gpu_shap = np.moveaxis(np.array(gpu_shap), [0, 1, 2], [2, 0, 1])
+    else:
+        gpu_shap = np.array(gpu_shap, copy=False)
     # Check outputs roughly the same as CPU algorithm
     assert np.allclose(ex.expected_value, gpu_ex.expected_value, 1e-3, 1e-3)
     assert np.allclose(host_shap, gpu_shap, 1e-3, 1e-3)
@@ -243,14 +248,7 @@ def test_gpu_tree_explainer_shap_interactions(task, feature_perturbation):
     ex = shap.GPUTreeExplainer(model, X, feature_perturbation=feature_perturbation)
     shap_values = np.array(ex.shap_interaction_values(X), copy=False)
 
-    assert (
-        np.abs(
-            np.sum(shap_values, axis=(len(shap_values.shape) - 1, len(shap_values.shape) - 2)).T
-            + ex.expected_value
-            - margin
-        ).max()
-        < 1e-4
-    ), "SHAP values don't sum to model output!"
+    assert np.allclose(np.sum(shap_values, axis=(1, 2)) + ex.expected_value, margin, atol=1e-4)
 
 
 @pytest.mark.xfail(reason="Categorical features not correctly implemented for GPU TreeSHAP")

